core/drand_daemon_control.go

```diff
@@ -197,6 +234,8 @@ func (dd *DrandDaemon) Stop(ctx context.Context) {
 			}
 		case <-t.C:
 			dd.log.Errorw("beacon process failed to terminate in 5 seconds, exiting forcefully", "id", bp.getBeaconID())
+			err := fmt.Errorf("beacon process %q failed to terminate in 5 seconds, exiting forcefully", bp.getBeaconID())
```

Comment:
could share the message between the log and `RecordError`

Reply:
Not really since the error message is formatted so we can query the logs by `id` field.

Reply:
Is there a reason not to use the formatted message in the log?